src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,139 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        try {
+            registrationService.register(null);
+        } catch (RuntimeException e) {
+            return;
+        }
+        fail("RuntimeException should be thrown when User is null");
+    }
+
+    @Test
+    void register_ageOver18User_Ok() {
+        user.setAge(25);
+        user.setLogin("Bob");
+        user.setPassword("rocket");
+        User actual = registrationService.register(user);
+        User expected = user;
+        assertTrue(expected.equals(actual), "User with age over 18 should be registered");
+
+    }
+
+    @Test
+    void register_age18User_Ok() {
+        user.setAge(18);
+        user.setLogin("Elise");
+        user.setPassword("rocket1981");
+        User actual = registrationService.register(user);
+        User expected = user;
+        assertTrue(expected.equals(actual), "User with age 18 should be registered");
+    }
+
+    @Test
+    void register_ageLower18User_notOk() {
+        user.setAge(17);
+        user.setLogin("Melody");
+        user.setPassword("rocket");
+        try {
+            registrationService.register(user);
+        } catch (RuntimeException e) {
+            return;
+        }
+        fail("RuntimeException should be thrown when age less then 18");
+    }
+
+    @Test
+    void register_negativeAgeUser_notOk() {
+        user.setAge(-1);
+        user.setLogin("Jane");
+        user.setPassword("rocket");
+        try {
+            registrationService.register(user);
+        } catch (RuntimeException e) {
+            return;
+        }
+        fail("RuntimeException should be thrown when age is negative");
```

Comment:
assertThrows(ExpectedException.class, () -> object.methodUnderTest(inputParams);

"[Try to avoid these common mistakes, while solving task](https://mate-academy.github.io/jv-program-common-mistakes/java-core/junit/registration-validation)"